packages/transport/src/api/udp.ts

```diff
@@ -33,13 +34,31 @@ export class UdpApi extends AbstractApi {
             if (!this.listening) return;
 
             this.enumerationTimeout = setTimeout(() => {
-                this.enumerate(this.enumerateAbortController.signal).finally(enumerateRecursive);
+                this.enumerate(this.enumerateAbortController.signal)
+                    .then(r => this.handleEnumerate(r))
+                    .finally(enumerateRecursive);
             }, 500);
         };
 
         enumerateRecursive();
     }
 
+    private handleEnumerate(result: Awaited<ReturnType<typeof this.enumerate>>) {
+        if (result.success) {
+            const [known, unknown] = arrayPartition(result.payload, device =>
+                this.devices.includes(device),
+            );
+
+            if (known.length !== this.devices.length || unknown.length > 0) {
```

Comment:
isn't uknown.length > 0 enough here?

Reply:
nope, it will not work for disconnect event, example:
i've got 1 device stored in `this.devices` now after disconnection i have 0 devices in known and 0 in unknown